core/src/main/resources/hudson/model/ManageJenkinsAction/index.jelly

```diff
@@ -85,9 +85,10 @@ THE SOFTWARE.
                   <a href="${m.urlName}">
                     <div class="jenkins-section__item__icon" aria-hidden="true">
                       <l:icon src="${m.iconFileName}" class="icon" />
-                      <j:if test="${m.getUpdateCount() != null and m.getUpdateCount() != 0}">
-                        <div tooltip="${m.getUpdateCount() == 1 ? '%updateAvailable(m.getUpdateCount())' : '%updatesAvailable(m.getUpdateCount())'}" class="jenkins-section__item__icon__badge">
-                          ${m.getUpdateCount()}
+                      <j:set var="badge" value="${m.badge}"/>
+                      <j:if test="${badge != null}">
+                        <div tooltip="${badge.tooltip}" class="jenkins-section__item__icon__badge alert-${badge.severity}">
```

Comment:
No `@NonNull` annotation on `severity`, and no fallback handling for `null` parameter, but I don't think the result looks reasonable when there's none.

(Untested, planning to check later unless someone else is faster.)

Reply:
severity is now non null